components/raftstore/src/store/fsm/peer.rs

```diff
@@ -2688,6 +2688,14 @@ where
             return Ok(());
         }
 
+        // As this peer is already stable (initialized and has gotten HeartBeats from
+        // the leader), it's safe to clear the pending uncampaigned regions.
+        if MessageType::MsgHeartbeat == msg_type
+            && !self.fsm.peer.uncampaigned_new_regions.is_empty()
+        {
+            self.fsm.peer.uncampaigned_new_regions.clear();
```

Comment:
Is there any chance that the peer clears the uncampaigned list before becoming the leader? Do we have any existing mechanism to prevent that? The event sequence I'm thinking of is: 
1. Peer 1 starts leader transfer.
2. Peer 2 applies the split and adds the new region to the uncampaigned list. 
3. Peer 2 receives a heartbeat from the leader and _clears_ the uncampaigned list. <==
4. Peer 2 becomes the leader. 
5. Peer 1 becomes a follower.
6. Peer 1 applies the split.

Reply:
Updated, the uncampaigned list in the follower will be cleared when it executes `CompactLog` to avoid this case.